rutil/ParseBuffer.cxx

```diff
@@ -671,25 +671,31 @@ ParseBuffer::integer()
 
    if (!isdigit(*mPosition))
    {
-       Data msg("Expected a digit, got: ");
-       msg += Data(mPosition, (mEnd - mPosition));
+      Data msg("Expected a digit, got: ");
+      msg += Data(mPosition, (mEnd - mPosition));
       fail(__FILE__, __LINE__,msg);
    }
-   
-   int num = 0;
-   int last=0;
-   while (!eof() && isdigit(*mPosition))
-   {
-      last=num;
-      num = num*10 + (*mPosition-'0');
-      if(last > num)
-      {
+
+   // The absolute value limit depending on the detected sign
+   const unsigned int absoluteLimit = negative ? -(unsigned int)INT_MIN : INT_MAX;
+   // maximum value for all full range expect last digit
+   const unsigned int border = absoluteLimit / 10;
+   // value the last digit must not exceed
+   const unsigned int digitLimit = absoluteLimit % 10;
+
+   unsigned int num = 0;
+   while (!eof() && isdigit(*mPosition)) {
```

Comment:
Should clean up bracketing style here, then merging this pull request is good.   Thanks!